jubilant/_juju.py

```diff
@@ -548,6 +552,19 @@ def wait(
             exc.add_note(str(status))
         raise exc
 
+    @functools.cached_property
+    def _juju_is_snap(self) -> bool:
+        which = shutil.which(self.cli_binary)
+        return which is not None and '/snap/' in which
+
+    @functools.cached_property
+    def _temp_dir(self) -> str:
+        if self._juju_is_snap:
+            # If Juju is running as a snap, we can't use /tmp, so put temp files here instead.
+            return os.path.expanduser('~/snap/juju')
```

Comment:
This just occurred to me: do we ever want to run jubilant with sudo?

Reply:
P.S. should this add `current` or `common`?

```
> ls ~/snap/juju/
29965/  30245/  common/  current@
```

Reply:
Per David's comment further down, I'm going to use `common` (and ensure it's created).